core/schema/modulesource.go

```diff
@@ -1061,6 +1061,9 @@ func callerHostFindUpContext(
 	bk *buildkit.Client,
 	curDirPath string,
 ) (string, bool, error) {
+	if !filepath.IsAbs(curDirPath) {
+		return "", false, fmt.Errorf("caller host path %q is not absolute", curDirPath)
+	}
```

Comment:
Ah okay, it's not strictly necessary, I put it here because a bug I fixed would have failed much earlier and more comprehensibly with this check in place.

I will remove it for now unless I can find some way of implementing it cross-platform. It's tbh quite annoying that the Go stdlib makes the `filepath` implementation for other platforms completely inaccessible unless you compile the entire binary for that platform... I'm sure there's some good reason for it though 😄

Reply:
At first glance, the implementation of `IsAbs` for windows is not too outrageous appearing: https://cs.opensource.google/go/go/+/refs/tags/go1.22.3:src/path/filepath/path_windows.go;l=59

So maybe we can just have our own util that does it? Requires the client tell us their platform, but that's extremely trivial plumbing.

Reply:
I just went with rm'ing the check for now, too much extra changes on top of an already large PR otherwise: https://github.com/dagger/dagger/commit/48e774a1e499482d67ae8c05760663b951751c95